containers/src/Kokkos_DynamicView.hpp

```diff
@@ -173,7 +173,8 @@ struct ChunkedArrayManager {
     void execute() {
       // Destroy the array of chunk pointers.
       // Two entries beyond the max chunks are allocation counters.
-      for (unsigned i = 0; i < m_chunk_max; i++) {
+      auto const len = *reinterpret_cast<size_t*>(m_chunks + m_chunk_max + 1);
```

Comment:
This feels so sketchy.  Why is it not something like
```suggestion
      size_t const len = *(m_chunks + m_chunk_max + 1);
```
Also what was wrong with `unsigned`?

Reply:
m_chunks is a pointer type that is abused to store a numerical value.

Reply:
https://github.com/kokkos/kokkos/blob/a1d045da4cc299217e2831f71dedff72e4f2aecf/containers/src/Kokkos_DynamicView.hpp#L356-L360